code/modules/vehicles/mecha/equipment/tools/mining_tools.dm

```diff
@@ -47,6 +47,10 @@
 			if(isliving(target))
 				drill_mob(target, source)
 				playsound(src,'sound/weapons/drill.ogg',40,TRUE)
+				// If we gibbed the target by drilling them, we can stop drilling them.
+				// Prevents starting a do_after on a qdeleted target.
+				if(QDELETED(target))
```

Comment:
move this underneath the return so it applies to objects as well

Reply:
I've tweaked some of the code in general, moving things around, adding some early returns and making sure the parent proc is returned (which manages the power drain and cooldown period) when the code reaches a point where the user has started the process of drilling.